lapis/src/main/antlr/org/genspectrum/lapis/model/advancedqueryparser/AdvancedQuery.g4

```diff
@@ -0,0 +1,131 @@
+grammar AdvancedQuery;
+
+// parser rules
+
+start: expr EOF;
+expr:
+  both             # Uni
+  | not_ expr         # Not
+  | expr and_ expr    # And
+  | expr or_ expr    # Or
+  | '(' expr ')'     # Parenthesis
+  | maybeQuery       # Maybe
+  ;
+
+both: 
+  single
+  | metadataQueryExpr
+  ;
+
+single:
+  nucleotideMutationQuery
+  | nOfQuery
+  | nucleotideInsertionQuery
+  | aaMutationQuery
+  | aaInsertionQuery
+  ;
+
+metadataQueryExpr:
+  metadataQuery
+  | metadataGreaterThanEqualQuery
+  | metadataLessThanEqualQuery
+  | regexMetadataQuery
+  | isNullQuery
+  ;
+
+variantExpr:
+  single # VariantUni
+  | not_ variantExpr # VariantNot
+  | variantExpr and_ variantExpr # VariantAnd
+  | variantExpr or_ variantExpr # VariantOr
+  | '(' variantExpr ')'  # VariantParenthesis
+  | maybe_ '(' variantExpr ')' # VariantMaybe
+  ;
+
+maybeQuery:
+  maybe_ '(' variantExpr ')'
+  ;
+
+or_: OR | '|';
+maybe_: M A Y B E;
+not_: NOT | '!';
+and_: AND | '&';
+position: NUMBER;
+
+nucleotideMutationQuery : nucleotideMutationQueryFirstSymbol? position nucleotideMutationQuerySecondSymbol?;
```

Comment:
Is this able to handle multisegmented organisms?

Reply:
I made the corresponding fixes - ended up being a largish change as we can't easily distinguish between an amino acid or segmented virus mutation query in advance as many of the ambiguous nucleotides and aa that we accept overlap

Reply:
Yes, I believe that's how it's supposed to be implemented, we also have that in the components. You parse a generic mutation, then you check from the sequence name whether is was a gene or a segment.